docs/agents/magpie.rst

```diff
@@ -0,0 +1,383 @@
+.. highlight:: rst
+
+.. _magpie:
+
+===============
+Magpie Agent
+===============
+
+The magpie is an incredibly intelligent bird, with a decent ability to mimic
+other bird calls, though not as good as `the superb lyrebird
+<https://www.youtube.com/watch?v=mSB71jNq-yQ>`_. In the context of OCS, the job
+of the Magpie agent is to take detector data from the SMuRF Streamer and
+translate into G3Frames that are compatible with lyrebird. This requires
+a small bit of pre-processing and downsampling that is done whenever a new
+frame is received. 
+
+.. argparse::
+   :filename: ../agents/magpie/magpie_agent.py
+   :func: make_parser
+   :prog: python3 magpie_agent.py
+
+Configuration File Examples
+------------------------------
+
+Below are example docker-compose and ocs configuration files for running the
+magpie agent.
+
+Site Config
+````````````
+Below is the site-config entry for the magpie instance we have running at UCSD
+on K2SO. We are displaying detectors using a wafer layout, determined by
+a detmap csv file, with a target sample rate of 20 Hz::
+
+      {'agent-class': 'MagpieAgent',
+       'instance-id': 'magpie-crate1slot2',
+       'arguments': [
+         '--stream-id', 'crate1-slot2',
+
+         '--src', 'tcp://localhost:4532',
+
+         '--dest', 8675,
+         '--delay', 5,
+         '--target-rate', 20,
+
+         '--layout', 'wafer',
+         # Detmap CSV file
+         '--det-map', '/home/jlashner/lyrebird_demo/detmap.csv',
+         '--offset', 0, 0,
+       ]},
+
+Docker
```

Comment:
For consistency with other Agent reference pages, please name this section "Docker Compose".